### ISSUE TITLE
All crates which rely on packaged libudev-sys fails with Permission denied

### ISSUE THREAD

**Issue by igor-raits** [dcuddeback/libudev-sys#2]
```
+ /usr/bin/cargo build --release -j8
   Compiling libc v0.2.34
   Compiling pkg-config v0.3.9
     Running `/usr/bin/rustc --crate-name libc /usr/share/cargo/registry/libc-0.2.34/src/lib.rs --crate-type lib --emit=dep-info,link -C opt-level=3 --cfg 'feature="default"' --cfg 'feature="use_std"' -C metadata=8f4c4fada3247809 -C extra-filename=-8f4c4fada3247809 --out-dir /home/brain/rpmbuild/BUILD/libudev-0.2.0/target/release/deps -L dependency=/home/brain/rpmbuild/BUILD/libudev-0.2.0/target/release/deps --cap-lints allow -Copt-level=3 -Cdebuginfo=2 -Clink-arg=-Wl,-z,relro,-z,now`
     Running `/usr/bin/rustc --crate-name pkg_config /usr/share/cargo/registry/pkg-config-0.3.9/src/lib.rs --crate-type lib --emit=dep-info,link -C opt-level=3 -C metadata=75dda3370d955f23 -C extra-filename=-75dda3370d955f23 --out-dir /home/brain/rpmbuild/BUILD/libudev-0.2.0/target/release/deps -L dependency=/home/brain/rpmbuild/BUILD/libudev-0.2.0/target/release/deps --cap-lints allow -Copt-level=3 -Cdebuginfo=2 -Clink-arg=-Wl,-z,relro,-z,now`
   Compiling libudev-sys v0.1.3
     Running `/usr/bin/rustc --crate-name build_script_build /usr/share/cargo/registry/libudev-sys-0.1.3/build.rs --crate-type bin --emit=dep-info,link -C opt-level=3 -C metadata=5843b4718211982e -C extra-filename=-5843b4718211982e --out-dir /home/brain/rpmbuild/BUILD/libudev-0.2.0/target/release/build/libudev-sys-5843b4718211982e -L dependency=/home/brain/rpmbuild/BUILD/libudev-0.2.0/target/release/deps --extern pkg_config=/home/brain/rpmbuild/BUILD/libudev-0.2.0/target/release/deps/libpkg_config-75dda3370d955f23.rlib --cap-lints allow -Copt-level=3 -Cdebuginfo=2 -Clink-arg=-Wl,-z,relro,-z,now`
     Running `/home/brain/rpmbuild/BUILD/libudev-0.2.0/target/release/build/libudev-sys-5843b4718211982e/build-script-build`
error: failed to run custom build command for `libudev-sys v0.1.3`
process didn't exit successfully: `/home/brain/rpmbuild/BUILD/libudev-0.2.0/target/release/build/libudev-sys-5843b4718211982e/build-script-build` (exit code: 101)
--- stdout
cargo:rustc-link-search=native=/usr/lib64
cargo:rustc-link-lib=udev

--- stderr
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Error { repr: Os { code: 13, message: "Permission denied" } }', src/libcore/result.rs:906:4
note: Run with `RUST_BACKTRACE=1` for a backtrace.
```

I guess this is because it tries to write something to directory where it is itself which is obviously read-only. It should write all stuff it needs only to `OUT_DIR` or such.

**Comment by dcuddeback** [comment]
@ignatenkobrain Thanks for the report. I'm unable to reproduce the problem, including with the latest stable and nightly compilers:

```
$ cargo build
   Compiling pkg-config v0.3.9
   Compiling libc v0.2.35
   Compiling libudev-sys v0.1.3
   Compiling libudev v0.2.0 (file:///home/david/src/rustlibs/libudev-rs)
    Finished dev [unoptimized + debuginfo] target(s) in 1.66 secs
```

Do you have more details on how to reproduce the issue?

**Comment by igor-raits** [comment]
@dcuddeback move libudev-sys to some non-writable location (e.g. `/usr/local/share`), make sure you are not able to write anything there, then change dependency in libudev to have `path = "/usr/local/share/libudev-sys"` instead of version and try to run `cargo build`.

**Comment by igor-raits** [comment]
If you are interested more in strace...

```
[...]
chdir("/usr/share/cargo/registry/libudev-sys-0.1.3") = 0
[...]
execve("/home/brain/rpmbuild/BUILD/libudev-0.2.0/target/release/build/libudev-sys-5843b4718211982e/build-script-build", ["/home/brain/rpmbuild/BUILD/libudev-0.2.0/target/release/build/libudev-sys-5843b4718211982e/build-script-build"], 0x7fcb1688e000 /* 103 vars */ <unfinished ...>
[...]
```

That script is creating `target/...` in relative path and given that it did `chdir()` and it obviously fails because that directory is not writable.

**Comment by igor-raits** [comment]
so what I'm saying is that you should avoid writing to relative paths and either use `OUT_DIR` or even create temporary files (tempfile crate?).

**Comment by dcuddeback** [comment]
@ignatenkobrain Thanks. That helped me reproduce it. I'll try making paths relative to `OUT_DIR`.

**Comment by igor-raits** [comment]
```rust
let out_dir = env::var_os("OUT_DIR").expect("can't get OUT_DIR");
let output = Path::new(&out_dir).join(test_file_name);
...
```

Something like this should work.

**Comment by dcuddeback** [comment]
@ignatenkobrain I just pushed up something very similar to a branch called `feature/fix-build-location`: https://github.com/dcuddeback/libudev-sys/commit/2e9798c915e56097692f1482d8aa4f6b76595331. Do you have time to verify that it fixes the problem you're seeing?

**Comment by igor-raits** [comment]
Yep, works fine! Could you merge & publish new version?

Thanks a lot !

**Comment by dcuddeback** [comment]
@ignatenkobrain Yeah. Thanks for testing the patch.

**Comment by dcuddeback** [comment]
@ignatenkobrain Published as v0.1.4. Thanks again for your help with this.
